internal/signal/signal.go

```diff
@@ -42,9 +42,6 @@ func newHaltSignalChan() chan os.Signal {
 //   the signal package.  Not ideal.
 func StopOnExitSignal(s Stopper) {
 	killSignals := newHaltSignalChan()
-
-	go func() {
-		<-killSignals
-		s.Stop()
-	}()
+	<-killSignals
+	s.Stop()
```

Comment:
I'm still unclear why we no longer do async handling of this logic

Reply:
Quoting from the clean history of my upcoming push :)

```
   Make StopOnExitSignal synchronous
    
    This will allow clients to decide if they want it to run async or not.
    Adding async is as simple as wrapping the call in a go routine.  Whereas
    it's useful to be able to block if we want to do, as we do in the main
    entrypoint method where Secretless needs to block after starting up all
    its services.
```

I'm open to doing this differently though, if you have a better solution.  Seems like something else we can think about and improve later if need be.  Personally I think the current solution is ok, but again am open to hearing reasons why it's not.

Reply:
So after thinking about this a while, I think the problems here if we move to a sync method are:
- there is most likely no longer a need for this method at all since the caller can just call `Stop()` right after
- there's two places that we use this in: one needs to wait (secretless) and one doesn't (entrypoint). we're catering to a one-off consumer with the sync logic. we can easily just add a "block indefinitely" block right after calling an async version in Secretless.
- if we do choose to leave it sync, it's no longer matching its name `StopOnExitSignal` - it would probably be something like `BlockUntilHalt`

Reply:
I went ahead and reverted `StopOnExitSignal` to being async... I think you're right.  It makes the client code cleaner and why not have it.

To deal with the blocking issue, I made `NewExitSignalChan` public, so we can use it like this after starting Secretless:

```go
	// Block until we receive an exit signal
	<- signal.NewExitSignalChan()
```